feature-libs/quote/components/quote-seller-edit/quote-seller-edit.component.service.spec.ts

```diff
@@ -0,0 +1,241 @@
+    it('should parse string', (done) => {
+      service.parseDiscountValue('100.00').subscribe((result) => {
+        expect(result).toBe(100);
+        done();
+      });
+    });
+
+    it('should consider locale specific decimal separator', (done) => {
+      service.parseDiscountValue('100.77').subscribe((result) => {
+        expect(result).toBe(100.77);
+        done();
+      });
+    });
+
+    it('should ignore locale specific grouping separator', (done) => {
+      service.parseDiscountValue('1,000.77').subscribe((result) => {
+        expect(result).toBe(1000.77);
+        done();
+      });
+    });
+  });
+
+  describe('getFormatter', () => {
+    it('should return a formatter for currency display ', (done) => {
+      service.getFormatter().subscribe((result) => {
+        expect(result.format(0)).toBe('$0.00');
+        done();
+      });
+    });
+  });
+
+  describe('checkAndReportCurrencyIfMissing', () => {
+    it('should throw error in case we do not find symbol or currency ISO code', () => {
+      expect(() =>
+        service['checkAndReportCurrencyIfMissing'](
+          'en',
+          new Intl.NumberFormat('en', {
+            style: 'currency',
+            currency: 'ALL',
+            currencyDisplay: 'narrowSymbol',
+          })
+        )
+      ).toThrowError();
+    });
+  });
+
+  describe('addTimeToDate', () => {
+    it('should add the local time to a given date', () => {
+      const dateWithTime = service.addTimeToDate(EXPIRATION_DATE_AS_STRING);
+      expect(dateWithTime).toContain(EXPIRATION_DATE_AS_STRING);
+      expect(dateWithTime).toContain('T');
+      expect(dateWithTime).toContain(TimeUtils.getLocalTimezoneOffset());
+    });
+  });
+
+  describe('removeTimeFromDate', () => {
+    it('should remove the time part from a time stamp', () => {
+      expect(service.removeTimeFromDate(EXPIRATION_TIME_AS_STRING)).toBe(
+        EXPIRATION_DATE_AS_STRING
+      );
+    });
+
+    it('should do nothing for undefined time stamp', () => {
+      expect(service.removeTimeFromDate(undefined)).toBeUndefined();
+    });
+  });
+
+  describe('performValidationAccordingToMetaData', () => {
+    it('should accept input using group and decimal separators', () => {
+      expect(
+        service['performValidationAccordingToMetaData'](
+          '1.000,76',
+          '.',
+          ',',
+          10
+        )
+      ).toBe(false);
+    });
+    it('should not accept input with 2 decimal separators', () => {
+      expect(
+        service['performValidationAccordingToMetaData'](
+          '1,000,76',
+          '.',
+          ',',
+          10
+        )
+      ).toBe(true);
+    });
+  });
+
+  describe('getMaximumNumberOfTotalPlaces', () => {
+    it('should compile number of total places from total if no quote discount is present, taking 2 decimal points into account ', () => {
+      expect(service.getMaximumNumberOfTotalPlaces(quote)).toBe(
+        Math.log10(TOTAL_PRICE) + 3
+      );
```

Comment:
Does calculating the expected value not just (partly) repeat the code under test? why not expect a  just '6'
```suggestion
      expect(service.getMaximumNumberOfTotalPlaces(quote)).toBe(6);
```

Reply:
Don't fully agree. Here I am just stating what I expect, having 6 would be ok but I would lose the reference to 'TOTAL_PRICE' (comment needed, but then if one changes the constant value one might not update the comment)

Reply:
than maybe 
```suggestion
      expect(service.getMaximumNumberOfTotalPlaces(quote)).toBe(TOTAL_PRICE.length + 2);
```
the `+ 2 ` also corresponds to the test description of `..., taking 2 decimal points into account`.

Reply:
Yes, agree

Reply:
Will also change in the other occurence